sc2/unit.py

```diff
@@ -295,6 +295,121 @@ def movement_speed(self) -> float:
         # TODO: use bot object to calculate if unit is zerg unit and if it is on creep, and if it your own unit: check for movement speed upgrades (e.g. zergling, hydra, ultralisk) or buffs (stimpack, time warp slow, fungal), perhaps write a second property function for this
         return self._type_data._proto.movement_speed
 
+    @property
+    def real_speed(self):
+        return self.calculate_speed()
+
+    def calculate_speed(self, upgrades=None) -> float:
+        """ Calculates the movement speed of the unit including buffs and upgrades.
+        Note: Upgrades only work with own units. Use "upgrades" param to set expected enemy upgrades.
+
+        :param upgrades: """
+        speed = self.movement_speed
+        unit_type = self.type_id
+
+        # ---- Upgrades ----
+        if upgrades is None:
+            upgrades = self._bot_object.state.upgrades
+
+        if upgrades:
```

Comment:
Perhaps this function could speed up if you do another if statement here, e.g.
`if upgrades and unit_type in {BANSHEE, OBSERVER, WARPPRISM, ...}:`
This way, Python doesn't check all the if / elif statements? Would have to benchmark that.